1. Cual es el valor del campo 'pin' del archivo 'U.S. Bank' para el registro 1?

In [1]:
head -n 2 U.S.\ Bank.txt | tail -n 1 | cut -d, -f4

"231"


***

2. Cuantos registros quedan en el archivo 'person' si se eliminan los registros con 'city' = 'Aurora (Colorado)'?

In [3]:
tail +2  person | cut -d, -f3 | wc -l

15000


In [2]:
tail +2  person | cut -d, -f3 | sed -n '/"Aurora (Colorado)"/p' | wc -l

142


15000 - 142 = 14858

***

3. Cual es la 'ciudad (estado)' en la posicion 88 del archivo 'person', si el archivo se organiza alfabeticamente por el campo 'ssn'?

In [7]:
tail +2  person | cut -d, -f1,3 | sort | head -n 88 | tail -1

"116-25-1276","Indianapolis[11] (Indiana)"
sort: write failed: 'standard output': Broken pipe
sort: write error


***

4. Cual es el valor del campo 'validthru' del archivo 'PNC' para el registro 3741?

In [6]:
tail +2 PNC.txt | cut -d, -f2 | head -n 3741 | tail -1

"Apr/2025"


***

5. Cuantos registros hay en el archivo 'person' para city = 'Lexington (Kentucky)'?

In [8]:
tail +2  person | cut -d, -f3 | sed -n '/"Lexington (Kentucky)"/p' | wc -l

163


In [7]:
grep 'Lexington (Kentucky)' person | wc -l

163


***

6. Cual el nombre completo (fullname) del del dueño de la tarjeta 3608-1395-4856-1915?

In [2]:
grep '3608-1395-4856-1915' *.csv | cut -d, -f2

"238-83-1981"


In [8]:
grep '238-83-1981' person | cut -d, -f5

"Mickey Shisler"


***

7. Cuantas personas nacieron en el trimestre Q4 del año?

In [ ]:
csvsql --query 'SELECT COUNT(*) FROM (SELECT * FROM person WHERE birthdate LIKE "%-10-%" UNION SELECT * FROM person WHERE birthdate LIKE "%-11-%" UNION SELECT * FROM person WHERE birthdate LIKE "%-12-%") as something' person

***

8. Cuantas veces aparece el número 4394 en el tercer grupo de digitos de las tarjetas de credito?

In [16]:
grep '....-....-4394-....' *.csv | wc -l

995


***

9. Cuantas personas tienen 'EA' como iniciales de su nombre?

In [17]:
tail +2 person | cut -d, -f5 | grep 'E.*A.*' | wc -l

23


***

10. Cuantas tarjetas tienen el pin entre 590 y 600?

In [2]:
cut -d, -f4 *.txt | grep '59[0-9]\|600' | wc -l

532


***

11. Cual es la cantidad maxima de tarjetas que tiene un banco?

In [3]:
wc -l *.csv |sort

   3798 American-Express.txt
   3811 Bank of America.txt
   3862 BarclayCard US.txt
   3797 Capital One.txt
   3838 Chase.txt
   3864 Citibank.txt
   3821 Discover.txt
   3931 MasterCard.txt
   3743 PNC.txt
   3861 USAA.txt
   3857 U.S. Bank.txt
   3870 Visa.txt
   3960 Wells Fargo.txt
  50013 total
